server/services/version.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { ObjectId } from 'mongodb'
+import { ApprovalStates } from '../models/Deployment'
+import '../utils/mockMongo'
+import VersionModel from '../models/Version'
+import { createVersion, findModelVersions, findVersionById, findVersionByName, serializedVersionFields } from './version'
+import UserModel from '../models/User'
+
+const modelId = new ObjectId
+const anotherModelId = new ObjectId
+
+const testVersion: any = {
+  model: modelId,
+  version: '1',
+  metadata: {},
+  built: false,
+  managerApproved: ApprovalStates.NoResponse,
+  reviewerApproved: ApprovalStates.NoResponse,
+  state: {},
+  logs: [],
+  createdAt: new Date(),
+  updatedAt: new Date()
+}
+
+const testVersion2: any = {
+  model: anotherModelId,
+  version: '1',
+  metadata: {},
+  built: true,
+  managerApproved: ApprovalStates.Accepted,
+  reviewerApproved: ApprovalStates.NoResponse,
+  state: {},
+  logs: [],
+  createdAt: new Date(),
+  updatedAt: new Date()
+}
+
+const testUser = {
+  userId: 'user1',
+  email: 'user1@email.com',
+  data: { some: 'value' },
+}
+const userDoc = new UserModel(testUser)
+
+describe('test version service', () => {
+
+  beforeEach(async () => {
+    const versionDoc = new VersionModel(testVersion)
+    await versionDoc.save()
+    const versionDoc2 = new VersionModel(testVersion2)
+    await versionDoc2.save()
+    testVersion._id = versionDoc._id
+  })
+
+  test('that the serializer returns the correct properties', () => {
+    const properties: any = serializedVersionFields()
+    expect(properties.mandatory).toStrictEqual(['_id', 'version', 'metadata.highLevelDetails.name'])
```

Comment:
I'm not a massive fan of any of these tests because, to me, I don't think they test anything.  I'd rather a unit test for the serializedVersionFields to try to serialize a `version` and make sure only those fields are available on it.